sklearn/metrics/tests/test_ranking.py

```diff
@@ -1052,3 +1080,23 @@ def test_ranking_loss_ties_handling():
     assert_almost_equal(label_ranking_loss([[1, 0, 0]], [[0.25, 0.5, 0.5]]), 1)
     assert_almost_equal(label_ranking_loss([[1, 0, 1]], [[0.25, 0.5, 0.5]]), 1)
     assert_almost_equal(label_ranking_loss([[1, 1, 0]], [[0.25, 0.5, 0.5]]), 1)
+
+
+def test_partial_roc_auc_score():
+    # Check `roc_auc_score` for max_fpr != `None`
+    y_true = np.array([0, 0, 1, 1])
+    assert_equal(roc_auc_score(y_true, y_true, max_fpr=1), 1)
+    assert_equal(roc_auc_score(y_true, y_true, max_fpr=0.001), 1)
+    assert_equal(np.isnan(roc_auc_score(y_true, y_true, max_fpr=0)), True)
```

Comment:
Just use
```python
assert np.isnan(roc_auc_score(y_true, y_true, max_fpr=0))
```

Reply:
But I would be in favor or returning 0 when max_fpr=0 (see my comment above). WDYT?